design-proposals/golden-image-delivery-and-update-pipeline.md

```diff
@@ -0,0 +1,207 @@
+
+## HCO: Integration with SSP Operator
+
+The HCO is responsible for setting VolumeImportPoller sources on the SSP operator’s CR. These volume sources are the locations used by the VolumeImportPollers to populate PVCs for the common templates.
+
+
+## OCP UI: Integration with VolumeReferences
+
+The UI currently detects if a template has a boot source by checking if the DataVolumeTemplate’s source PVC is present. The UI needs to now be capable of checking if a DataVolumeTemplate’s source VolumeReference has phase: linked as well in order to determine if a boot source is set.
+
+## Scalability
+
+The only scaling concern is that the automated process has the potential to exhaust storage resources unless a PVC garbage collection mechanism is observed by the VolumeImportPoller.
+
+## Update/Rollback Compatibility
+
+* SSP operator must make image pipelines an opt-in feature to ensure existing boot images are not overwritten. 
+* UI must be capable of detecting boot sources for DataVolumeTemplates with sources for both VolumeReferences and PVCs
+
+## Implementation Phases
+
+**Phase1** - CDI APIs
+* VolumeReference API/controller
+* VolumeImportPoller API/controller
+
+**Phase2** - SSP + Common Templates
+* Common Templates transition to VolumeReferences for DataVolumeTemplate source
+* SSP Operator creates VolumeImportPollers for automatic PVC import
+
+**Phase3** - HCO and UI
+* HCO Team integrates with SSP Operator’s API for setting volume import sources
+* UI Team transitions to detecting boot source using a linked VolumeReference rather than looking directly for a PVC of a specific name.
+
+
+
+# API Examples
+## VM + DataVolumeTemplate + VolumeReference as source
+
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: DataVolume
+metadata:
+   name: my-fedora-vm-image
+spec:
+   pvc:
+     accessModes:
+     - ReadWriteOnce
+     resources:
+       requests:
+         storage: 2Gi
+   source:
+     volumeReference:
+       name: fedora
+       namespace: some-namespace
+```
+
+## VolumeImportPoller Example
+
+This example polls for image tags for the quay.io/kubevirt/fedora image and imports the latest image sorting tags based on semver. As new tags are introduced, the VolumeImportPoller will import them as new PVCs and automatically manage updating the corresponding VolumeReference.
+
+* **source** specifies where to poll from
+* **pollInterval** specifies how often to look for new imports 
+* **garbageCollectOutdated** specifies whether old PVCs should be cleaned up after a new PVC is imported.
+
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: VolumeImportPoller
+metadata:
+   name: fedora-poller
+spec:
+   source:
+     registry:
+       url: docker://quay.io/kubevirt/fedora
+    pollInterval: 12h
+    garbageCollectOutdated: true
+status:
+  managedVolumeReference:
```

Comment:
status could hold a "lastImportedPVC" field
Also, should we be able to point VolumeImportPollers at references manually so we can create refs but maintain them manually and/or add a poller later if the admin/we want to?

Reply:
> status could hold a "lastImportedPVC" field

i like that. I'm adding it to the doc. 

> Also, should we be able to point VolumeImportPollers at references manually so we can create refs but maintain them manually and/or add a poller later if the admin/we want to?

yeah, i changed my thinking around this after you brought up the issue with how SSP should transition from opt-out to opt-in of this auto-update feature. I think the poller should be able to adopt ownership of existing VolumeReferences and that reference should be specified on teh poller's api.  I've updated the doc to reflect this.